core/unix/os.c

```diff
@@ -3686,9 +3695,9 @@ dr_create_client_thread(void (*func)(void *param), void *arg)
      * the thread list for the app, making it more invisible.
      */
     uint flags = CLONE_VM | CLONE_FS | CLONE_FILES |
-        CLONE_SIGHAND IF_NOT_X64(| CLONE_SETTLS)
-        /* CLONE_THREAD required.  Signals and itimers are private anyway. */
-        IF_VMX86(| (os_in_vmkernel_userworld() ? CLONE_THREAD : 0));
+        CLONE_SIGHAND
+            /* CLONE_THREAD required.  Signals and itimers are private anyway. */
```

Comment:
Without CLONE_SETTLS we do not need to create and pass &desc, so that code can go, right.

Reply:
Looks like yes. It mentions 'Linux' and VMX86_SERVER in the comment. But I think this is just with regards to differences setting the tls in clone. So I guess it is hopefully ok even though I didn't want to touch it. Done.

Reply:
Well, you can't remove CLONE_SETTLS and leave desc: without CLONE_SETTLS that &desc parameter is ignored by the kernel and it is not clean code to pass it in.

Reply:
(I didn't want to touch the entire CLONE_SETTLS but I think it's ok.)